api/v1alpha1/packagebundle.go

```diff
@@ -45,11 +46,27 @@ func (config *PackageBundle) FindSource(pkgName, pkgVersion string) (retSource P
 	return retSource, fmt.Errorf("package not found in bundle (%s): %s @ %s", config.ObjectMeta.Name, pkgName, pkgVersion)
 }
 
+func (config *PackageBundle) IsNewer(pkgBundle *PackageBundle) bool {
```

Comment:
I had the same thought on not comparing major-minor, but I think it would be odd not to. We should probably add a version here sometime like matchesKubeVersion(version) and return true/false, but I don't want to do that in this PR

Reply:
I created https://github.com/aws/eks-anywhere-packages/issues/137 to track the invalid name issue. It is too much for this PR. I would like to address your "LessThan" comment though, I like that idea.

Reply:
bundle k8s version matching issue https://github.com/aws/eks-anywhere-packages/issues/138

Reply:
Done